cirq/iqm/iqm_device_test.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+import cirq
+
+
+def test_valid_operations():
+    adonis = cirq.iqm.Adonis()
+
+    adonis.validate_operation(cirq.GateOperation(cirq.X,
+                                                 [cirq.GridQubit(0, 1)]))
+
+    adonis.validate_operation(
+        cirq.GateOperation(cirq.YPowGate(exponent=0.25),
+                           [cirq.GridQubit(0, 1)]))
+
+    adonis.validate_operation(
+        cirq.GateOperation(
+            cirq.CZ,
+            [cirq.GridQubit(1, 0), cirq.GridQubit(1, 1)]))
+
+    adonis.validate_operation(
+        cirq.GateOperation(
+            cirq.ISwapPowGate(exponent=0.5),
+            [cirq.GridQubit(1, 0), cirq.GridQubit(1, 1)]))
+
+
+def test_invalid_operations():
+    adonis = cirq.iqm.Adonis()
+
+    with pytest.raises(ValueError):
+        adonis.validate_operation(
+            cirq.GateOperation(cirq.H, [cirq.GridQubit(0, 1)]))
+
+    with pytest.raises(ValueError):
+        adonis.validate_operation(
+            cirq.GateOperation(cirq.CNOT,
```

Comment:
It is shorter and clearer to using gate.on().  I would also make the qubits variables, since they are used multiple times.  For instance,

```
q1_0 = cirq.GridQubit(1,0)
q1_1 = cirq.GridQubit(1,1)
adonis.validate_operation(cirq.H(q1_0))
adonis.validate_operation(cirq.CNOT(q1_0,q1_1))
```

Reply:
Now using this style.